clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleNamedImageLookupController.java

```diff
@@ -73,31 +69,41 @@ private GoogleNamedImageLookupController(Cache cacheView) {
       @RequestParam(required = false) String q,
       @RequestParam(required = false) String account,
       HttpServletRequest request) {
-    Map<String, List<Image>> imageMap = listImagesByAccount();
-
-    // Set of accounts for which we should return images: either the supplied account, or default
-    // to all accounts
-    Set<String> accounts;
-    if (StringUtils.isNotEmpty(account)) {
-      accounts = new HashSet<>();
-      if (imageMap.containsKey(account)) {
-        accounts.add(account);
-      }
-    } else {
-      accounts = imageMap.keySet();
-    }
+    Collection<CacheData> imageCacheData = getImageCacheData(account);
+    Predicate<NamedImage> queryFilter = getQueryFilter(q);
+    Predicate<NamedImage> tagFilter = getTagFilter(request);
+    return imageCacheData.stream()
+        .map(this::createNamedImageFromCacheData)
+        .filter(queryFilter)
+        .filter(tagFilter)
+        .sorted(Comparator.comparing(image -> image.imageName))
+        .collect(Collectors.toList());
+  }
 
-    List<NamedImage> results = new ArrayList<>();
-    for (String imageAccount : accounts) {
-      for (Image i : imageMap.get(imageAccount)) {
-        Map<String, Object> attributes = new HashMap<>();
-        attributes.put("creationDate", i.get("creationTimestamp"));
-        NamedImage newImage =
-            new NamedImage(imageAccount, i.getName(), attributes, buildTagsMap(i));
-        results.add(newImage);
-      }
+  private Collection<CacheData> getImageCacheData(String account) {
+    // If no account supplied, return images from all accounts
+    String pattern =
+        account == null
```

Comment:
I'm not sure if this is a significant change, but prior to this change we'd return images for all accounts when the account is the empty string, whereas now we only return all accounts when it's null.

Reply:
Another great catch for @ezimanyi :baseball: Adding `.orEmpty` check!